### PR TITLE
Make ASyncEntity handle XPathSyntax exceptions

### PR DESCRIPTION
**Pull request by orangejenny**
evaluate now throws an XPathSyntaxException

### CODE REVIEW

**Comment by orangejenny** [comment]
cross-requested with https://github.com/dimagi/commcare/pull/58

File: app/src/org/commcare/android/models/AsyncEntity.java [hunk 1]
```diff
@@ -115,6 +115,9 @@ public Object getField(int i) {
                 } catch(XPathException xpe) {
                     xpe.printStackTrace();
                     data[i] = "<invalid xpath: " + xpe.getMessage() + ">";
+                } catch (XPathSyntaxException xse) {
```
Comment (ctsims) [inline app/src/org/commcare/android/models/AsyncEntity.java]: Question: We don't generally handle these here because we try to parse all expressions inbound when we install the app, allowing people to catch syntax errors (which should be detectable before runtime) early. Is it impossible to make sure the graph expressions are compiled during install to throw their errors up there?

Comment (orangejenny) [inline app/src/org/commcare/android/models/AsyncEntity.java]: Good point, it'd be better to catch these errors earlier. Will do a separate PR.
